learn-regular-expressions-by-building-a-spam-filter/index.js

```diff
@@ -0,0 +1,40 @@
+const messageInput = document.getElementById("message-input");
+const result = document.getElementById("result");
+const checkMessageButton = document.getElementById("check-message-btn");
+
+/**
+ * I read through the previous discussions for this project
+ * and we can start with matches for literal phrases and then have the
+ * campers create more complex regex which seemed to be the original intent for this project.
+ *
+ *
+ *
+ */
+
+const blacklistRegexps = [
+  /[0-9] hundred|thousand|million|billion dollars/gi,
```

Comment:
Currently this works well. It matches "thousand" in "100 thousand dollars".

If we want to show how to match everything in "100 million dollars",  we could build this line up to something like `\d*\s+(hundred|thousand|million|billion)(\s+dollars)?`.

It could start off exactly like it is now, then get a bit more complex by adding a capturing groups and using character classes like `\d` to replace `[0-9]`, and special flags like `*` and `?`.

`\s` is probably not necessary, and can be replaced with a whitespace character. I just find `\s` a bit easier to read.

Reply:
We could add some steps that refactors this ` /[0-9] hundred|thousand|million|billion dollars/gi` to slowly introduce the `\d`, `*` and `?` flags. 
I can add a comment in the codebase about this.